assets/javascripts/discourse/initializers/hook-decrypt-post.js.es6

```diff
@@ -371,106 +371,121 @@ export default {
+        ) {
+          state.encrypted = ciphertext;
+
+          if (!window.isSecureContext) {
+            state.decrypting = false;
+            state.decrypted = true;
+            state.error = I18n.t("encrypt.preferences.insecure_context");
           } else {
-            attrs.cooked =
-              "<div class='alert alert-error'>" +
-              iconHTML("times") +
-              " " +
-              state.error +
-              "</div>" +
-              attrs.cooked;
+            state.decrypting = true;
+            getIdentity().then((identity) => {
+              if (!identity) {
+                // Absence of private key means user did not activate encryption.
+                showModal("activate-encrypt", { model: this });
+                return;
+              }
+
+              getTopicKey(topicId)
+                .then((key) => decrypt(key, ciphertext))
+                .then((plaintext) => {
+                  if (plaintext.signature) {
+                    getDebouncedUserIdentity(plaintext.signed_by_name)
+                      .then((userIdentity) => {
+                        return verify(
+                          userIdentity.signPublic,
+                          plaintext,
+                          ciphertext
+                        );
+                      })
+                      .then((result) => {
+                        verified[attrs.id] = checkMetadata(attrs, plaintext);
+                        if (!result) {
+                          verified[attrs.id].push({
+                            attr: "signature",
+                            actual: false,
+                            expected: true,
+                          });
+                        }
+                      })
+                      .catch(() => {
+                        verified[attrs.id] = [
+                          {
+                            attr: "signature",
+                            actual: false,
+                            expected: true,
+                          },
+                        ];
+                      })
+                      .finally(() => this.scheduleRerender());
+                  }
+
+                  return cookAsync(plaintext.raw);
+                })
+                .then((cooked) => (state.decrypted = cooked.string))
+                .catch(() => {
+                  state.decrypted = true;
+                  state.error = I18n.t("encrypt.decryption_failed");
+                })
+                .finally(() => {
+                  state.decrypting = false;
+                  this.scheduleRerender();
+                });
+            });
           }
+        }
 
-          return this._super(...arguments);
-        },
-      });
+        if (state.decrypted && state.decrypted !== true) {
+          attrs.cooked = state.decrypted;
+          Ember.run.next(() => {
+            const $post = $(`article[data-post-id='${attrs.id}']`);
```

Comment:
I think small action posts are `div`s, not `article`s. Is anything broken here? I do not know, but mentions and hashtags may not work as expected.

Reply:
nice catch! You were right, mentions in small actions were not working - it was plain text. Added fix to collect correct element :)